packages/format-currency/package.json

```diff
@@ -0,0 +1,28 @@
+{
+	"name": "@automattic/format-currency",
+	"version": "1.0.0",
+	"description": "JavaScript library for formatting currency",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
+	"repository": {
+		"type": "git",
+		"url": "git://github.com/Automattic/wp-calypso.git"
+	},
+	"keywords": [
+		"currency",
+		"internationalization"
+	],
+	"author": "Automattic, Inc.",
+	"license": "GPL-2.0-or-later",
+	"bugs": {
+		"url": "https://github.com/Automattic/wp-calypso/issues"
+	},
+	"homepage": "https://github.com/Automattic/wp-calypso",
+	"dependencies": {
+		"i18n-calypso": "file:../i18n-calypso"
+	},
+	"scripts": {
+		"build": "node ../../bin/build-package",
+		"clean": "npx rimraf dist"
+	}
```

Comment:
I _think_ this is side effect free, should we mark it as such? @blowery 

> A "side effect" is defined as code that performs a special behavior when imported, other than exposing one or more exports. An example of this are polyfills, which affect the global scope and usually do not provide an export.

https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Reply:
Sounds like a good idea. Code looks fairly side-effect free to me -- the one thing I'm not totally sure about would be the `i18n-calypso` import: https://github.com/Automattic/wp-calypso/blob/6dbbfc4aa0655968325f91eae7c680a9cd037e25/packages/format-currency/src/index.js#L4

Reply:
(More generally asking: In order to mark a package as side-effect free, do all it's dependencies also need to be side-effect free?)

Reply:
I'm curious about that myself…

It seems like a package-aware bundler would be smart enough to identify which packages and sub packages are side effect free (or not) and do the right thing.

That said: `i18n-calypso` itself is side effect free, so we're likely safe here. I'll open a PR.

 https://github.com/Automattic/wp-calypso/blob/6dbbfc4aa0655968325f91eae7c680a9cd037e25/packages/i18n-calypso/package.json#L7

Reply:
#31345